### PR TITLE
Issue #247: Added unit tests for LEDHelper functions

### LINKED ISSUE #247: Add Unit Tests for LEDHelper Class
**Description:**

We need to add unit tests for the LEDHelper class in the com.opensourcewithslu.outputdevices package. This is a "good-first-issue" because it involves creating unit tests, which is a manageable task for newcomers and essential for ensuring code reliability.

**Tasks:**

1. Setup Testing Framework:
    * Ensure that the project is configured to use JUnit and Mockito for unit testing.
    * Add necessary dependencies to build.gradle.
2. Create Unit Tests:
    * Test ledOn() Method:
        * Verify that the LED is turned on when it is currently off.
        * Ensure that the LED is not turned on if it is already on.
    * Test ledOff() Method:
        * Confirm that the LED is turned off when it is currently on.
        * Ensure that the LED is not turned off if it is already off.
    * Test switchState() Method:
        * Verify that the LED state is toggled correctly when it is currently on.
        * Confirm that the LED state is toggled correctly when it is currently off.
    * Test blink(int duration) Method:
        * Ensure that the blink method is called with the correct parameters.
3. Verify Logging:
    * Ensure that the appropriate log messages are generated for the ledOn() and ledOff() methods.
   
**Acceptance Criteria:**

* All unit tests for the LEDHelper class are added and passing.
* The tests should mock the DigitalOutput class to avoid actual hardware interaction.
* Ensure that all methods are tested for different scenarios as outlined

### PR DESCRIPTION
**Pull request by leandrumartin**
Fixes #247. Added JUnit and Mockito dependencies in in the `pi4micronaut-utils` folder to allow unit tests in that folder. Added unit tests for `LEDHelper` class functions to ensure they work, mocking the `DigitalOutput` class since the testing does not involve physical Raspberry Pis. All the tests pass.

The issue description adds that the tests should check whether the `LEDHelper` generates appropriate log messages, but this is not yet taken care of.

### CODE REVIEW

**Review by Ed0827 — COMMENTED** [review]
It looks great! All the unit tests for the LEDHelper class functions work as intended, and the conditional statements are descriptive and easy to follow.

File: pi4micronaut-utils/src/test/java/com/opensourcewithslu/Pi4MicronautUtilsTest.java
Comment (yrlmanoharreddy) [inline pi4micronaut-utils/src/test/java/com/opensourcewithslu/Pi4MicronautUtilsTest.java:1]: The tests address key functionality but don't follow the modular structure we discussed, which is crucial for consistency and maintainability. From a product perspective, aligning the tests with the project's design ensures future scalability. The tests cover important scenarios, but they need to follow the same structure as the main code. Please revise accordingly to improve clarity and maintainability. Great effort—looking forward to the updated tests!

**Comment by yrlmanoharreddy** [comment]
The tests address key functionality but don't follow the modular structure we discussed, which is crucial for consistency and maintainability. From a product perspective, aligning the tests with the project's design ensures future scalability. The tests cover important scenarios, but they need to follow the same structure as the main code. Please revise accordingly to improve clarity and maintainability. Great effort—looking forward to the updated tests!

**TASK:**
**Changes that need to be done:**

"Follow the structure same as the main code's structure."

Example in your case:

* Actual main code structure:
pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/LEDHelper.java
* Expected Structure for unit test should be
pi4micronaut-utils/src/test/java/com/opensourcewithslu/outputdevices/LEDHelperTest.java

**Review by yrlmanoharreddy — CHANGES_REQUESTED** [review]
The tests address key functionality but don't follow the modular structure we discussed, which is crucial for consistency and maintainability. From a product perspective, aligning the tests with the project's design ensures future scalability. The tests cover important scenarios, but they need to follow the same structure as the main code. Please revise accordingly to improve clarity and maintainability. Great effort—looking forward to the updated tests!

TASK:
Changes that need to be done:

"Follow the structure same as the main code's structure."

Example in your case:

Actual main code structure:
pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/LEDHelper.java
Expected Structure for unit test should be
pi4micronaut-utils/src/test/java/com/opensourcewithslu/outputdevices/LEDHelperTest.java
Additionally, please use "camelCase" conventions for member functions.

**Review by yrlmanoharreddy — CHANGES_REQUESTED** [review]
The tests address key functionality but don't follow the modular structure we discussed, which is crucial for consistency and maintainability. From a product perspective, aligning the tests with the project's design ensures future scalability. The tests cover important scenarios, but they need to follow the same structure as the main code. Please revise accordingly to improve clarity and maintainability. Great effort—looking forward to the updated tests!

TASK:
Changes that need to be done:

"Follow the structure same as the main code's structure."

Example in your case:

Actual main code structure:
pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/LEDHelper.java
Expected Structure for unit test should be
pi4micronaut-utils/src/test/java/com/opensourcewithslu/outputdevices/LEDHelperTest.java
Additionally, please use "camelCase" conventions for member functions.

**Review by yrlmanoharreddy — CHANGES_REQUESTED** [review]
The test cases are functional but currently don’t adhere to the modular structure we discussed, which is essential for maintaining consistency and ensuring long-term scalability. From a product standpoint, aligning the test structure with the main codebase will make future expansion and maintenance smoother. While the tests cover critical scenarios, they need to be organized to reflect the structure of the main code.

Required changes:

1. Follow the same structure as the main code:
    Here’s an example:
    
    **Current structure:** pi4micronaut-utils/src/main/java/com/opensourcewithslu/LEDHelper.java
    **Main code structure**: pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/LEDHelper.java
    **Expected structure for the unit test**: pi4micronaut-utils/src/test/java/com/opensourcewithslu/outputdevices/LEDHelperTest.java
    Please ensure your tests follow the same structure as the main codebase.

2. Two of your test cases are failing:
ledOnLogsWhenTurnedOn()
ledOffLogsWhenTurnedOff()
Kindly fix these issues before pushing the changes back

**Comment by leandrumartin** [comment]
I believe I've made all the necessary changes now.

**Review by yrlmanoharreddy — APPROVED** [review]
Done the changes according to the review and resolved issue #252 and also resolved merge conflicts in this commit

**Pull request merged**